enterprise_access/apps/api/v1/views/customer_billing.py

```diff
@@ -181,37 +199,136 @@ def create_checkout_session(self, request, *args, **kwargs):
 
     @extend_schema(
         tags=[CUSTOMER_BILLING_API_TAG],
-        summary='Create a new Customer Portal Session.',
+        summary='Create a new Customer Portal Session from the Admin portal MFE.',
     )
     @action(
-        detail=True,
+        detail=False,
         methods=['get'],
-        url_path='create-portal-session',
+        url_path='create-enterprise-admin-portal-session',
+    )
+    # # UUID in path is used as the "permission object" for role-based auth.
+    @permission_required(
+        CUSTOMER_BILLING_CREATE_PORTAL_SESSION_PERMISSION,
+        fn=lambda request, **kwargs: request.GET.get('enterprise_customer_uuid') or kwargs.get(
+            'enterprise_customer_uuid')
```

Comment:
@iloveagent57 , @pwnage101  - Sanity check: The fn in `@permission_required` now pulls `enterprise_customer_uuid` from `request.GET` (query param) instead of only `kwargs`, so RBAC actually evaluates the right enterprise UUID instead of `None` (which caused blanket 403s before).

Reply:
ahh, right, that makes sense.